### PR TITLE
authorship view for dashboard + support for local zipped report

### PR DESCRIPTION
**Pull request by ongspxm**
```
blank for now
```

transient PR that adds in support for authorship view and support for local zipped report.

### CODE REVIEW

**Comment by damithc** [comment]
What's this PR about?

**Comment by ongspxm** [comment]
@eugenepeh @AdityaA1998 @yong24s @yamidark ready for review

File: frontend/src/static/js/api.js [hunk 1]
```diff
@@ -1,53 +1,51 @@
...
+    var err = () => window.alert("unable to get file");
+
+    if(window.REPORT_ZIP){
+        window.REPORT_ZIP.file(file.slice(1)).async("text")
+            .then((txt) => fn(JSON.parse(txt)));
+    }else{
+        var xhr = new XMLHttpRequest();
+        xhr.open("GET", file);
+        xhr.onload = function() {
+            if (xhr.status === 200) {
```
Comment (yong24s) [inline frontend/src/static/js/api.js]: Inconsistency with usage of space

File: template/static/js/v_authorship.js [hunk 2]
```diff
@@ -0,0 +1,153 @@
+window.toggleNext = function toggleNext(ele) {
```
[the remark this answers is not in the document]
Comment (ongspxm) [reply template/static/js/v_authorship.js]: :-1: not really sure what codacy is complaining about :(

File: template/static/js/api.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
+/* util funcs */
```
[the remark this answers is not in the document]
Comment (ongspxm) [reply template/static/js/api.js]: :-1: not sure what is it talking about too, it a comment...

File: template/static/js/main.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+window.REPORT_DIR = "";
```
[the remark this answers is not in the document]
Comment (ongspxm) [reply template/static/js/main.js]: :confused: no brackets?

File: template/static/js/main.js [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      this.tabAuthor = obj.author;
+      this.tabRepo = obj.repo;
+
+      this.isTabActive = true;
+      this.isTabAuthorship = true;
+    },
+  },
+  components: {
+    v_summary: window.vSummary,
+    v_authorship: window.vAuthorship,
```
[the remark this answers is not in the document]
Comment (ongspxm) [reply template/static/js/main.js]: nothing changing this to align with the html tag for vue

**Comment by ongspxm** [comment]
sorry guys, i messed up the rebase... rebasing one more time

**Comment by ongspxm** [comment]
@yamidark @yong24s @AdityaA1998 @eugenepeh checked and ready to go

@codacy-bot, imma ignore you

File: src/main/java/reposense/report/RepoInfoFileGenerator.java [hunk 6]
```diff
@@ -85,12 +83,12 @@ private static void generateIndividualRepoReport(RepoConfiguration repoConfig, L
                 Constants.STATIC_INDIVIDUAL_REPORT_TEMPLATE_ADDRESS);
```
Comment (yamidark) [inline src/main/java/reposense/report/RepoInfoFileGenerator.java]: The `Path templateLocation` above is no longer used, can be removed.

**Comment by yamidark** [comment]
@eugenepeh for your review

**Comment by ongspxm** [comment]
Will remove when mereging frontend with master, alongside all the java side
rebasing

**Pull request merged**